sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/_converters/_ai_services.py

```diff
@@ -0,0 +1,468 @@
+        """
+        Breaks a list of messages into query and response messages based on the run ID.
+
+        :param messages: The list of messages to be broken into query and response.
+        :type messages: List[Message]
+        :param run_id: The ID of the run to distinguish response messages.
+        :type run_id: str
+        :return: A tuple containing two lists - the first list contains query messages, and the second list contains response messages.
+        :rtype: (List[Message], List[Message])
+        """
+        query: List[Message] = [what for what in messages if what.run_id != run_id]
+        responses: List[Message] = [what for what in messages if what.run_id == run_id]
+        return query, responses
+
+    @staticmethod
+    def __filter_messages_up_to_run_id(chronological_messages, run_id: str):
+        """
+        Filters messages up to a specific run ID.
+
+        :param chronological_messages: The list of messages in chronological order.
+        :type chronological_messages: List[Message]
+        :param run_id: The ID of the run.
+        :type run_id: str
+        :return: The filtered list of messages.
+        :rtype: List[Message]
+        """
+        filtered_messages = []
+        in_my_current_run = False
+        for single_turn in chronological_messages:
+            # Since this is the conversation of the entire thread and we are interested in a given run, we need to
+            # filter out the messages that came after the run.
+            if single_turn.run_id is not None:
+                if single_turn.run_id == run_id:
+                    in_my_current_run = True
+
+            # Then, if we think that we are currently in our run and we have a message that is not from our run,
+            # it means that we have left our run.
+            if in_my_current_run and single_turn.run_id != run_id:
+                break
+
+            # We're good to add it.
+            filtered_messages.append(single_turn)
+
+        return filtered_messages
+
+    @staticmethod
+    def __is_agent_tool_call(message: Message) -> bool:
+        """
+        Determines if a message is an agent tool call.
+
+        :param message: The message to be checked.
+        :type message: Message
+        :return: True if the message is an agent tool call, False otherwise.
+        :rtype: bool
+        """
+        return (
+            message.role == _AGENT  # Any other agent that this run's.
+            and isinstance(message.content, list)  # Content is of expected type.
+            and len(message.content) > 0  # There are messages/calls/results present.
+            and "type" in message.content[0]  # Being safe here.
+            and message.content[0]["type"] == _TOOL_CALL  # Not interested in assistant's toolcalls.
+        )
+
+    @staticmethod
+    def __sort_messages(messages: List[Message]) -> List[Message]:
```

Comment:
Any limitation of order param in list messages assistant API, that needs us to do this ?

Reply:
Not limitation there, but since we make two kinds of API calls to fulfill the final list of messages:

1. Getting regular messages with `list_messages` (in chronological order)
2. Getting tool call details with `list_run_steps` (in chronological order)

When we concatenate the two lists, we need to rebuild the chronological view with the two combined.